src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -76,16 +77,26 @@ default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName
          * https://javadoc.jenkins-ci.org/jenkins/tasks/SimpleBuildStep.html */
         
         String tmpDir = getNodeSpecificTmpFolderPath(workspace);
-        return new FilePath(launcher.getChannel(), tmpDir+"/"+uniqueName);
+        String fileSeperator = "/";
+
+        if (!launcher.isUnix()){
+            fileSeperator = "\\";
+        }
+
+        return new FilePath(launcher.getChannel(), tmpDir + fileSeperator + uniqueName);
     }
 
     default String getNodeSpecificTmpFolderPath(FilePath workspace) throws IOException, InterruptedException {
         Computer cmp = workspace.toComputer();
         if (cmp == null) {
             throw new IOException(Message.getValue("build.workspace.computer.not.found"));
         }
-        String tmpDir = (String) cmp.getSystemProperties().get("java.io.tmpdir");
-        return tmpDir;
+        
+        String tmpDirPath = (String) cmp.getSystemProperties().get("java.io.tmpdir");
+
+        // Invoke FilePath.normalize for clean file path.
+        FilePath tmpDir = new FilePath(new File(tmpDirPath));

```

Comment:
I would recommend to test the behaviour using remote agent once since we are using File() object which represents the local file .

Reply:
I tweaked implementation a bit to get channel from computer and create a FilePath. I confirmed this works both on master and remote agents.

On Master :
![jenkinOnMaster](https://user-images.githubusercontent.com/68549146/89151203-0aae1d80-d57e-11ea-8a2f-d6a5900d10e4.PNG)

On Remote Agent :
![jenkinOnRemoteAgent](https://user-images.githubusercontent.com/68549146/89151206-0bdf4a80-d57e-11ea-882e-1ae326470812.PNG)